src/daemon.rs

```diff
@@ -1,8 +1,10 @@
 use crate::errors::SdError;
 use libc::pid_t;
 use nix::unistd;
+use std::os::unix::io::RawFd;
 use std::os::unix::net::UnixDatagram;
 use std::{env, fmt, fs, path, time};
+use uds::UnixDatagramExt;
```

Comment:
Unfortunatly looking at the implementation of send_fds and C implementations, not without considerable effort and the use of unsafe.

Using low level socket stuff isn't trivial and tricky to get to work properly. Also reimplementation of common crates causes bloat since multiple different implementations with the same purpose end up in the binaries, and it masks any CVEs found in crates when using e.g. cargo audit. So I think that would be a rather bad idea in this case.

Reply:
Well, it looks like some of the stuff is in nix. But I personally don't feel very confident in implementing this kind of thing. A lot of this stuff seem rather archaic and I'd rather not get it wrong.

But I tried compiling the regular notify example with and without this change and the binary size stay the same so rustc is able to remove any of the extra code if the regular notify function is used. And the uds crate has only one non-optional dependency which is libc so it shouldn't carry much risk.